### ISSUE TITLE
MySQL stacktrace after paper updated their mysql-connector-java to 8.0.23

### ISSUE THREAD

**Issue by Erlpil** [pop4959/LWCX#148]
**LWC Version (/lwc admin version):** 
2.2.6-2d10269
**Server version (/version):** 
Paper-590 (MC: 1.16.5)
**Short five word summary of issue:** 
MySQL stacktrace after paperjar update.

**Detailed description of problem:**
We jumped from paper version 582 to 590.
Then booting and performing a /lwc admin cleanup it spits out a stacktrace that did not happen before the paper update.
I think it can be related to: https://github.com/PaperMC/Paper/commit/d560151ecdad2eabccda797883c9e699b0985104
We have several servers running with LWC and mySQL, and some of them add more info than others.

**If there was an error in console, put it here:**
Server 1:
```
On boot:
[07:52:49] [Server thread/INFO]: [LWC] Enabling LWC v2.2.6-2d10269
[07:52:49] [Server thread/INFO]: [LWC] Connecting to MySQL
[07:52:49] [Server thread/WARN]: Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
[07:52:49] [Server thread/INFO]: [LWC] Performing database optimization for 1.13...
[07:52:49] [Server thread/INFO]: [LWC] Performing database migration to 1.13 format...
[07:52:49] [Server thread/WARN]: com.griefcraft.scripting.ModuleException: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.sql.Database.printException(Database.java:196)
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateAquatic(PhysDB.java:2357)
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:426)
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1726)
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:401)
[07:52:49] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[07:52:49] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[07:52:49] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[07:52:49] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[07:52:49] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[07:52:49] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:252)
[07:52:49] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055)
[07:52:49] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[07:52:49] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[07:52:49] [Server thread/WARN]: Caused by: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347)
[07:52:49] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1025)
[07:52:49] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateAquatic(PhysDB.java:2348)
[07:52:49] [Server thread/WARN]: 	... 12 more
```

```
On /lwc admin cleanup
[08:47:01] [Server thread/INFO]: You have: 15604 protections. 
[08:47:01] [Server thread/INFO]: This may take a while depending on how many protections you have
[08:47:01] [Thread-41/INFO]: Processing cleanup request now in a separate thread
[08:47:01] [Thread-41/INFO]: [LWC] Performing database optimization for 1.13...
[08:47:01] [Thread-41/INFO]: [LWC] Performing database migration to 1.13 format...
[08:47:01] [Thread-41/INFO]: [LWC] Exception caught during cleanup: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[08:47:01] [Thread-41/WARN]: com.griefcraft.scripting.ModuleException: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[08:47:01] [Thread-41/WARN]: 	at com.griefcraft.sql.Database.printException(Database.java:196)
[08:47:01] [Thread-41/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateAquatic(PhysDB.java:2357)
[08:47:01] [Thread-41/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:426)
[08:47:01] [Thread-41/WARN]: 	at com.griefcraft.modules.admin.AdminCleanup$Admin_Cleanup_Thread.run(AdminCleanup.java:177)
[08:47:01] [Thread-41/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[08:47:01] [Thread-41/WARN]: Caused by: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347)
[08:47:01] [Thread-41/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1025)
[08:47:01] [Thread-41/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateAquatic(PhysDB.java:2348)
[08:47:01] [Thread-41/WARN]: 	... 3 more

// Does not do the cleanup
```

Server 2:
```
On boot:
[07:59:49] [Server thread/INFO]: [LWC] Enabling LWC v2.2.6-2d10269
[07:59:49] [Thread-21/INFO]: [AsyncWorldEdit] You have the latest version of the plugin.
[07:59:49] [Server thread/INFO]: [LWC] Connecting to MySQL
[07:59:49] [Server thread/WARN]: Loading class `com.mysql.jdbc.Driver'. This is deprecated. The new driver class is `com.mysql.cj.jdbc.Driver'. The driver is automatically registered via the SPI and manual loading of the driver class is generally unnecessary.
[07:59:49] [Server thread/INFO]: [LWC] Performing database optimization for 1.13...
[07:59:50] [Server thread/INFO]: [LWC] Unable to add new block to lwc_blocks!
[07:59:50] [Server thread/WARN]: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1025)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:154)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:174)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateLWCX(PhysDB.java:2286)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:425)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1726)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:401)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:252)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[07:59:50] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[07:59:50] [Server thread/INFO]: [LWC] Unable to add new block to lwc_blocks!
[07:59:50] [Server thread/WARN]: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1025)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:154)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:174)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateLWCX(PhysDB.java:2286)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:425)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1726)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:401)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:252)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[07:59:50] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[07:59:50] [Server thread/INFO]: [LWC] Unable to add new block to lwc_blocks!
[07:59:50] [Server thread/WARN]: java.sql.SQLIntegrityConstraintViolationException: Duplicate entry '1' for key 'PRIMARY'
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:117)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:953)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1092)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdateInternal(ClientPreparedStatement.java:1040)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeLargeUpdate(ClientPreparedStatement.java:1347)
[07:59:50] [Server thread/WARN]: 	at com.mysql.cj.jdbc.ClientPreparedStatement.executeUpdate(ClientPreparedStatement.java:1025)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:154)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:174)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateLWCX(PhysDB.java:2286)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:425)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1726)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:401)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:252)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[07:59:50] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[07:59:50] [Server thread/WARN]: java.lang.IllegalArgumentException: Name cannot be null
[07:59:50] [Server thread/WARN]: 	at org.apache.commons.lang.Validate.notNull(Validate.java:192)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.Material.matchMaterial(Material.java:4667)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.Material.matchMaterial(Material.java:4650)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.cache.BlockCache.addBlock(BlockCache.java:172)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.doUpdateLWCX(PhysDB.java:2286)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.sql.PhysDB.load(PhysDB.java:425)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWC.load(LWC.java:1726)
[07:59:50] [Server thread/WARN]: 	at com.griefcraft.lwc.LWCPlugin.onEnable(LWCPlugin.java:401)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPlugin.setEnabled(JavaPlugin.java:263)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.java.JavaPluginLoader.enablePlugin(JavaPluginLoader.java:380)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.plugin.SimplePluginManager.enablePlugin(SimplePluginManager.java:483)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugin(CraftServer.java:501)
[07:59:50] [Server thread/WARN]: 	at org.bukkit.craftbukkit.v1_16_R3.CraftServer.enablePlugins(CraftServer.java:415)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.DedicatedServer.init(DedicatedServer.java:252)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.w(MinecraftServer.java:1055)
[07:59:50] [Server thread/WARN]: 	at net.minecraft.server.v1_16_R3.MinecraftServer.lambda$a$0(MinecraftServer.java:289)
[07:59:50] [Server thread/WARN]: 	at java.base/java.lang.Thread.run(Thread.java:834)
[07:59:54] [Server thread/WARN]: [LWC] "LWC v2.2.6-2d10269" has registered a listener for com.palmergames.bukkit.towny.event.TownUnclaimEvent on method "public void com.griefcraft.modules.pluginsupport.Towny.onTownUnclaim(com.palmergames.bukkit.towny.event.TownUnclaimEvent)", but the event is Deprecated. "Server performance will be affected"; please notify the authors [Hidendra, pop4959, Me_Goes_RAWR].
```

**Comment by pop4959** [comment]
Thanks for reporting this, as I wasn't aware that Paper changed it. I can't look into it immediately, but I will see if I can do anything about this later today.

**Comment by pop4959** [comment]
Should be fixed in b8700aca561ba4688d301df34a8600bb04e6aa71 now by shading in our own version of the MySQL connector. Everything appears to be working again on my test server, but please try for yourself and let me know if you have any further issues. The latest dev builds can be found [here](https://ci.codemc.io/view/Author/job/pop4959/job/LWCX-dev/).
